src/mococrw/mac.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+#include <memory>
+#include <vector>
+#include "openssl_wrap.h"
+
+namespace mococrw
+{
+
+/**
+ * @brief The MessageAuthenticationCode class is the abstract base class for the different implementation of
+ * message authentication codes.
+ */
+class MessageAuthenticationCode {
+public:
+    /**
+     * Adds the message to the hash. This function may be invoked multiple times.
+     * For finishing the calculation of the MAC-tag invoke finish().
+     *
+     * @param message chunk of data used for MAC.
+     * @throws MoCOCrWException if this function is invoked after finish was called.
+     */
+    void update(const std::vector<uint8_t>& message);
+
+    /**
+     * @brief Finalize the HMAC.
+     *
+     * This function calculates the hashed messaged authentication code.
```

Comment:
```suggestion
     * This function calculates the message authentication code.
```

"hashed message authentication code" may be mis-interpreted as calculating the MAC and hashing the generated MAC. Let's avoid this ambiguity and just call it "message authentication code".